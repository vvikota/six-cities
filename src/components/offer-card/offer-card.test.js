import React from 'react';
import renderer from 'react-test-renderer';
import OfferCard from './offer-card.jsx';

const mock = {
  premium: true,
  price: 120,
  image: `img/apartment-01.jpg`,
  rating: 3,
  placeType: `Apartment`,
  placeDiscription: `Wood and stone place`
};

it(`OfferCard is correcctly render`, ()=> {

  const tree = renderer.create(<OfferCard
    offer={mock}
    openCard={jest.fn()}
    onMouseEnter={jest.fn()}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
