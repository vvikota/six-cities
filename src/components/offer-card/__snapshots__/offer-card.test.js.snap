// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard is correcctly render 1`] = `
<article
  className="near-places__card place-card"
  onMouseEnter={[MockFunction]}
>
  <div
    className="place-card__image-wrapper near-places__image-wrapper"
  >
    <a
      className="image-link"
      href="#"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://es31-server.appspot.com/six-cities/static/hotel/6.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          250
        </b>
        <span
          className="place-card__price-text"
        >
          /Â night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            href="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "64%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        className="place-card__name-link"
        href="/offer/1"
        onClick={[Function]}
      >
        Amazing and Extremely Central Flat
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      room
    </p>
  </div>
</article>
`;
